LoopKit/InsulinKit/PumpEvent+CoreDataClass.swift

```diff
@@ -162,6 +234,18 @@ extension PumpEvent {
             value = entry.value
             unit = entry.unit
             deliveredUnits = entry.deliveredUnits
+            modelDuration = entry.insulinModel?.effectDuration
+            modelDelay = entry.insulinModel?.delay
+            // ANNA TODO: could this be more elegant?
+            if let model = entry.insulinModel as? ExponentialInsulinModel {
+                modelPeak = model.peakActivityTime
+                modelType = .exponential
+            } else if let model = entry.insulinModel as? ExponentialInsulinModelPreset {
+                modelPeak = (model.getExponentialModel() as! ExponentialInsulinModel).peakActivityTime
+                modelType = .exponential
+            } else if let _ = entry.insulinModel as? WalshInsulinModel {
+                modelType = .walsh
+            }
```

Comment:
Not sure if there's a better way to assign (or structure) the persisted insulin model data. I currently have 4 fields that are used to recreate the model: the type of model (exponential, walsh, or none), the peak time (if exponential), the DIA, and the delay.

Reply:
Same comments apply here re moving to using InsulinModelSettings as the representation.